### PR TITLE
Fix Build fails if quarkus-oidc-client is not in the classpath

### LINKED ISSUE #824: Build fails if `quarkus-oidc-client` is not in the classpath
As part of the build generation introduced in #795, the latest release now breaks projects that are not using OIDC:

```
[ERROR] Failed to execute goal com.redhat.quarkus.platform:quarkus-maven-plugin:3.8.4.redhat-00002:build (default) on project openapi-bug: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkiverse.openapi.generator.deployment.GeneratorProcessor#additionalBean threw an exception: java.lang.NoClassDefFoundError: io/quarkus/oidc/client/runtime/AbstractTokensProducer
[ERROR]         at java.base/java.lang.ClassLoader.defineClass1(Native Method)
[ERROR]         at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
[ERROR]         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:508)
[ERROR]         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:468)
[ERROR]         at io.quarkiverse.openapi.generator.deployment.GeneratorProcessor.additionalBean(GeneratorProcessor.java:72)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:840)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.lang.ClassNotFoundException: io.quarkus.oidc.client.runtime.AbstractTokensProducer
[ERROR]         at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:42)
[ERROR]         at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:225)
[ERROR]         at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:210)
[ERROR]         at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:205)
[ERROR]         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:518)
[ERROR]         at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:468)
[ERROR]         ... 16 more
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

```

This openAPI spec doesn't even have `security` tags, so all the authz classes should be skipped and not generated.

### PR DESCRIPTION
**Pull request by rmanibus**
Fix #824 

Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [ ] Pull Request contains link to the issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-quarkus2` to backport the original PR to the `quarkus2` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by rmanibus** [comment]
@ricardozanini I managed to reproduce it by not using the parent for this new integration test

File: client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/GeneratorProcessor.java [hunk 1]
```diff
@@ -61,6 +64,10 @@ void additionalBean(
             Capabilities capabilities,
             BuildProducer<AdditionalBeanBuildItem> producer) {
 
+        if (!isClassPresentAtRuntime(ABSTRACT_TOKEN_PRODUCER)) {
```
Comment (rmanibus) [inline client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/GeneratorProcessor.java:67]: the oidc client does not declare any capability, I need to use this as a fallback

**Comment by rmanibus** [comment]
not sure why it is not able to find the parent pom on the CI, I got it to work on my machine

File: client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/GeneratorProcessor.java (hunk 1 above)
Comment (rmanibus) [reply client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/GeneratorProcessor.java:67]: https://quarkusio.zulipchat.com/#narrow/channel/187038-dev/topic/Default.20Capability.20For.20extension

**Comment by ricardozanini** [comment]
> not sure why it is not able to find the parent pom on the CI, I got it to work on my machine

Are you running:

```
mvn '-Dorg.slf4j.simpleLogger.log.org.openapitools=off' -B formatter:validate impsort:check verify --file pom.xml -DskipTests
```

From the root?

**Comment by rmanibus** [comment]
hey @ricardozanini I finally got some time to work on this. Off course, I had to nuke my m2 folder to reproduce the relative path error, that's why I was not able to see it on my machine. 
Anyway, it's fixed now. Do you want to release this now or should we wait for the capability to be backported in 3.15.2 ?

**Comment by rmanibus** [comment]
if you want to merge this now I can provide a new PR once the new release is out !

**Comment by ricardozanini** [comment]
@rmanibus, we can merge this now, release it (because the main is somewhat broken), and backport it to the new-to-be LTS branch. WDYT?

**Review by ricardozanini — APPROVED** [review]

**Comment by hbelmiro** [comment]
@ricardozanini FYI, we can use the [backport-pr workflow](https://github.com/quarkiverse/quarkus-openapi-generator/blob/main/.github/workflows/pr-backporting.yml) to backport it to other branches.

**Pull request merged**